pkg/hub/custommetrics/custommetrics.go

```diff
@@ -0,0 +1,198 @@
+var managedClusterMetric = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	Name: "a_managed_cluster",
+	Help: "Managed Cluster being managed by ACM Hub.",
+}, []string{"cluster_name", "cluster_id", "type", "version", "cluster_infrastructure_provider", "hub_id"})
+
+func getDynClient() (dynamic.Interface, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return dynamic.NewForConfig(config)
+}
+
+func getStaticData(c dynamic.Interface) {
+
+	cvObj, errCv := c.Resource(cvGVR).Get(context.TODO(), "version", metav1.GetOptions{})
+	if errCv != nil {
+		klog.Fatalf("Error getting cluster version %v \n", errCv)
+		panic(errCv.Error())
+	}
+	cv := &ocinfrav1.ClusterVersion{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(cvObj.UnstructuredContent(), &cv)
+	if err != nil {
+		klog.Fatalf("Error unmarshal cluster version object%v \n", err)
+		panic(errCv.Error())
+	}
+	hubID = string(cv.Spec.ClusterID)
+	klog.Infof("Hub id of this ACM Server is " + hubID)
+
+}
+
+func addCluster(obj interface{}) {
+	klog.Info("Adding Managed Clusters ####################")
+	j, err := json.Marshal(obj.(*unstructured.Unstructured))
+	if err != nil {
+		klog.Warning("Error on ManagedCluster marshal.")
+	}
+	managedCluster := clusterv1.ManagedCluster{}
+	err = json.Unmarshal(j, &managedCluster)
+	if err != nil {
+		klog.Warning("Error on ManagedCluster unmarshal.")
+
+	}
+
+	klog.Infof("Managed Cluster name being added: %s \n", managedCluster.GetName())
+
+	/* cluster := &clusterv1.ManagedCluster{}
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(mc.UnstructuredContent(), &cluster)
+	if err != nil {
+		klog.Fatalf("Error unmarshal managed cluster object%v \n", err)
+	} */
+
+	//TODO:
+	//get the actual values as mentioned here:
+	//https://github.com/open-cluster-management/perf-analysis/blob/master/Big%20Picture.md#acm-20-telemetry-data
+	managedClusterMetric.WithLabelValues(managedCluster.GetName(), "cluster_id", "type", "version", managedCluster.GetLabels()["cloud"], hubID).Set(1)
+}
+
+func delCluster(obj interface{}) {
+	klog.Info("Deleting Managed Clusters ####################")
+
+	j, err := json.Marshal(obj.(*unstructured.Unstructured))
+	if err != nil {
+		klog.Warning("Error on ManagedCluster marshal.")
+	}
+	managedCluster := clusterv1.ManagedCluster{}
+	err = json.Unmarshal(j, &managedCluster)
+	if err != nil {
+		klog.Warning("Error on ManagedCluster unmarshal.")
+	}
+
+	klog.Infof("Managed Cluster name being removed: %s \n", managedCluster.GetName())
+
+	//TODO:
+	//get the actual values as mentioned here:
+	//https://github.com/open-cluster-management/perf-analysis/blob/master/Big%20Picture.md#acm-20-telemetry-data
+	managedClusterMetric.WithLabelValues(managedCluster.GetName(), "cluster_id", "type", "version", managedCluster.GetLabels()["cloud"], hubID).Set(0)
+
+}
+
+func fetchTestData(c dynamic.Interface) {
```

Comment:
needs to be cleaned up before merge.

Reply:
I am hoping you meant a proper name and not fetchTestData. If so, it has the correct name now.